scripts/getData.py

```diff
@@ -1,135 +1,121 @@
+import string, os, time, json, re, unicodedata, html.parser, requests
 
 # Bon appetit cafe hours api url
-url 	   = "http://legacy.cafebonappetit.com/api/2/cafes?cafe="
+url = "http://legacy.cafebonappetit.com/api/2/cafes"
 # How many cafeterias you want to parse (in order)
-totalCafes = 1374
+totalCafes = 1818
 # What our file should be named
 fileName = "data.json"
 
 
 # Our constructed JSON data
 responseData = []
 # Parser to clean up the string names
-h = HTMLParser.HTMLParser()
+h = html.parser.HTMLParser()
 # Time when we started the script
 start = time.time()
 
-def appendData( cafeId, cafeName, cafeLoc ):
-	responseData.append( {'id':cafeId,'label':cafeName, 'desc':cafeLoc} )
+def appendData(cafeId, cafeName, cafeLoc):
+	print(cafeId + ") " + cafeName + " in " + cafeLoc)
+	responseData.append({'id':cafeId, 'label':cafeName, 'desc':cafeLoc})
 
 
-def cleanUpTitle( stringToClean ):
+def cleanUpTitle(stringToClean):
 	# Remove beginning and ending whitespace
 	string = stringToClean.strip()
 	# Replace html
-	cleanString = h.unescape( string )
+	cleanString = h.unescape(string)
 	# Replace unicode
-	cleanString2 = unicodedata.normalize( 'NFKD', cleanString ).encode( 'ascii','ignore' )
-	# Capitalize string
-	return cleanString2[0].capitalize() + cleanString2[1:]
+	cleanString2 = unicodedata.normalize('NFKD', cleanString)
+	return cleanString2
 
 
-def cleanUpLocation( stringToClean ):
+def cleanUpLocation(stringToClean):
 	# Remove beginning and ending whitespace
 	string = stringToClean.strip()
 	# Replace html
-	cleanString = h.unescape( string )
-	# Replace unicode
-	cleanString2 = unicodedata.normalize( 'NFKD', cleanString ).encode( 'ascii','ignore' )
-
-	# Handle city/state (length will be longer than 2 chars)
-	if len( cleanString2 ) > 2:
-		# Capitalize each first letter
-		string2 = cleanString2.title()
-	else:
-		string2 = cleanString2[0].capitalize() + cleanString2[1:]
-
-	return string2
+	cleanString = h.unescape(string)
+	# Replace unicodecleanString
+	cleanString2 = unicodedata.normalize('NFKD', cleanString)
+	return cleanString2
 
 
 # Finds the cafeteria id and name
-def getBonAppMenuData( url, id ):
-	# Construct the full url
-	url2 = ''.join( [url, str(id)] )
-
-   	# Receive the content of url and convert it to a string
-	response  = urlopen( url2 )
-	data 	  = str( response.read() )
+def getBonAppMenuData(url, id):
+	params = {'cafe': id}
+	response = requests.get(url, params=params)
+	data = response.text
 
 	try:
 		# Grab cafe Id from JSON
-		cafeId    = json.loads( data )["cafes"].keys()[0]
+		cafeId = list(json.loads(data)["cafes"].keys())[0]
 
 		# Grab cafe name from JSON
-		cafeName  = json.loads( data )["cafes"][cafeId]["name"]
+		cafeName = json.loads(data)["cafes"][cafeId]["name"]
 
 
 		# We want to display titles at least. We can do without location.
 		try:
 			# Grab cafe city from JSON
-			cafeCity  = json.loads( data )["cafes"][cafeId]["city"]
+			cafeCity = json.loads(data)["cafes"][cafeId]["city"]
```

Comment:
Use dict.get(key, default) instead of dict[key] to avoid the try/except blocks, and remove these try/excepts.